src/renderer/widgets/WalletDetails/model/wallet-provider-model.ts

```diff
@@ -134,37 +129,74 @@ const $signatoryWallets = combine(
   },
 );
 
-const $proxyWalletForProxied = combine(
+const $chainsProxies = combine(
+  {
+    accounts: $accounts,
+    chains: networkModel.$chains,
+    proxies: proxyModel.$proxies,
+  },
+  ({ accounts, chains, proxies }): Record<ChainId, ProxyAccount[]> => {
+    const proxiesForAccounts = uniqBy(accounts, 'accountId').reduce<ProxyAccount[]>((acc, account) => {
+      if (proxies[account.accountId]) {
+        acc.push(...proxies[account.accountId]);
+      }
+
+      return acc;
+    }, []);
+
+    const chainsMap = Object.keys(chains).reduce<Record<ChainId, ProxyAccount[]>>((acc, chainId) => {
+      acc[chainId as ChainId] = [];
+
+      return acc;
+    }, {});
+
+    return proxyUtils.sortAccountsByProxyType(proxiesForAccounts).reduce((acc, proxy) => {
+      acc[proxy.chainId].push(proxy);
+
+      return acc;
+    }, chainsMap);
```

Comment:
`groupBy` maybe used here

Reply:
I need to preserve original chains order